### DISCUSSION TITLE
Comment like/dislike API enhancement

### DISCUSSION THREAD

**Discussion by jdbit** [danirus/django-comments-xtd#351] [Ideas]
Hi,
I'm trying to implement vanilla JS ajax comments (without heavy jQuery and ReactJS dependencies) and have troubles with `django_comments_xtd.api.serializers.FlagSerializer` as it returns only a comment ID and flag, so I can't update likes and dislikes count based on this response. Even if I increment likes count based on a non-empty response, I can't be sure if dislikes should be decremented or not (for example, if the user disliked the comment earlier). Any chances likes and dislikes count will be added to this serializer in the future? Or maybe there is a workaround?
Thank you!

**Comment by danirus** [comment]
If you don't need the ReactJS plugin, I suggest you to look at the `rel-3.0.0` branch. It doesn't use any frontend framework. Also the model CommentFlag is replaced with a `CommentReaction` model that allows for better querying user reactions. There is already a working JavaScript module `static/django_comments_xtd/js/reactions.js`. You can see it action with the new demo project provided in the branch. I recently answered [here](https://github.com/danirus/django-comments-xtd/issues/349#issuecomment-948291053) on how to setup the demo project.

I still need to add vanilla JS to post comments. 
As for the rest, the `rel-3.0.0` is in better shape than `master`. 
Take a look and let me know if you have any question.

**Comment by danirus** [comment]
This is how the new demo project looks like.
By default reactions are only "Like" and "Dislike". But the demo project extends them to resemble the reactions allowed here, in GitHub comments.

<img width="1440" alt="Screenshot 2021-10-26 at 18 17 22" src="https://user-images.githubusercontent.com/157103/138919728-5a95782a-c12d-4a12-a7b2-295a7a9208ab.png">

**Comment by jdbit** [comment]
Wow, looks nice, I like it!
